sru/src/main/java/whelk/sru/servlet/SruServlet.java

```diff
@@ -0,0 +1,140 @@
+package whelk.sru.servlet;
+
+import org.apache.cxf.staxutils.StaxUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import whelk.Document;
+import whelk.JsonLd;
+import whelk.Whelk;
+import whelk.converter.marc.JsonLD2MarcXMLConverter;
+import whelk.exception.InvalidQueryException;
+import whelk.search2.*;
+import whelk.util.WhelkFactory;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.*;
+
+public class SruServlet extends HttpServlet {
+    private final Logger logger = LogManager.getLogger(this.getClass());
+
+    Whelk whelk;
+    JsonLD2MarcXMLConverter converter;
+    XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+    VocabMappings vocabMappings;
+    ESSettings esSettings;
+
+    public void init() {
+        whelk = WhelkFactory.getSingletonWhelk();
+        converter = new JsonLD2MarcXMLConverter(whelk.getMarcFrameConverter());
+        vocabMappings = new VocabMappings(whelk);
+        esSettings = new ESSettings(whelk);
+    }
+
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
+
+        // Same as the old version
+        res.setCharacterEncoding("utf-8");
+        res.setContentType("text/xml");
+
+        Map<String, String[]> parameters = req.getParameterMap();
+
+        if ( !parameters.containsKey("operation") || !parameters.containsKey("query") ) {
+            logger.debug("Bad SRU query: " + parameters);
+            res.sendError(400);
+        }
+
+        if ( parameters.get("operation").length != 1 || !parameters.get("operation")[0].equals("searchRetrieve") ) {
+            logger.debug("Bad SRU query (operation/searchRetrieve expected): " + parameters);
+            res.sendError(400);
+        }
+
+        String queryString = parameters.get("query")[0];
+        String instanceOnlyQueryString = "(" + queryString + ") AND type=Instance";
+
+        Map<String, Object> results;
+        try {
+            // This part is a little weird
+            HashMap<String, String[]> paramsAsIfSearch = new HashMap<>();
+            String[] q = new String[]{instanceOnlyQueryString};
+            paramsAsIfSearch.put("_q", q);
```

Comment:
```suggestion
            paramsAsIfSearch.put("_q", q);
            paramsAsIfSearch.put("_stats", new String[] { "false" }); // don't need facets
```

can save some work in elastic here

Reply:
Could maybe wrap all of this in a static method of `Query`? Something like `Query.simpleQuery(String q, ...)` (no filters, facets etc.)